### PR TITLE
Badges

### LINKED ISSUE #9605: [Suggestion] Prosecutors Badge
For making all of the criminals who want rp to lose hope.

Make the badge look similar to the prosecutors badge in investigations.

### PR DESCRIPTION
**Pull request by AnthonyJNgo**
:cl: AnthonyJNgo
add: Adds three new badges. An attorneys badge, a security badge, and a command badge.
/:cl:

[why]: 

Attorneys badge was already passed by head approval, adding a security and command badge was just a matter of changing the icon and description for each one.

Includes three new icons, three new objects, and a unique suicide message. You're able to attach the badge in the belt slot, and you're able to flash it to anyone nearby.

badgesonground

badgeflavortext

Fixes #9605

### CODE REVIEW

**Comment by ghost** [comment]
make it attachable to your uniform, similar to the attorney badge, otherwise +1

File: hippiestation/code/game/objects/items/badges.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	throwforce = 3
+	w_class = WEIGHT_CLASS_TINY
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced")
+
+/obj/item/badge/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to stab [user.p_their()] eyes with \the [src]! The responsibility was too much for them! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return BRUTELOSS
+
+/obj/item/badge/attack_self(mob/user) //All badges will flash/shine the same way.
+	user.visible_message("[user] flashes their shiny badge.", "<span class='notice'>You briefly flash your badge.</span>")
```
Comment (k-k-karma) [inline hippiestation/code/game/objects/items/badges.dm]: spam time

**Comment by DaydreamPuppeh** [comment]
My big issue is the sprite

**Comment by yoyobatty** [comment]
Why would you include adding a .dm file as part of the changelog?

**Comment by AnthonyJNgo** [comment]
> Why would you include adding a .dm file as part of the changelog?

I thought this was the github changelog, not the ingame one. It's a brand new file so I felt it was relevant in case anyone wants to expand off it.

Code is G2G though right? 

> My big issue is the sprite

art was never my strong point, I can try and redo it or ask around to see if someone can make it better.

**Comment by yoyobatty** [comment]
Nope, this is for the ingame one, just remove it.

File: hippiestation/code/game/objects/items/badges.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/obj/item/clothing/accessory/badge //I've moved this under accessories so they can be placed on jumpsuits or on the belt slot.
+	name = "badge"
+	desc = "Does badge things."
+	icon = 'hippiestation/icons/obj/badge.dmi'
+	slot_flags = ITEM_SLOT_BELT //You can attach this in the belt slot. Ideally ID would go in PDA and be placed in ID slot, and this could be put in belt slot.
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
+	resistance_flags = FIRE_PROOF | ACID_PROOF //IDs are immune to fire and acid, badges should be also.
+	force = 3 //weaker than a toy, but it doesn't make sense for it to do no damage.
+	throwforce = 3
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced")
```
Comment (yoyobatty) [inline hippiestation/code/game/objects/items/badges.dm:19]: pretty sure there are defines for these

**Comment by DaydreamPuppeh** [comment]
Remember to rename this attorney's badge to prosecutor's badge. There is already an attorneys badge ingame

File: hippiestation/code/game/objects/items/badges.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+/obj/item/clothing/accessory/badge/suicide_act(mob/living/carbon/user) //somewhat unique suicide message for fun.
+	user.visible_message("<span class='suicide'>[user] begins to stab [user.p_their()] eyes with \the [src]! The responsibility was too much for them! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return BRUTELOSS
+
+/obj/item/clothing/accessory/badge/attack_self(mob/user) //All badges will flash/shine the same way.
+	user.visible_message("[user] flashes their shiny badge.", "<span class='notice'>You briefly flash your badge.</span>")
```
Comment (GuyonBroadway) [inline hippiestation/code/game/objects/items/badges.dm]:
```suggestion
	user.visible_message("[user] flashes their shiny badge.", "<span class='notice'>You briefly flash your [name].</span>")
```
So it says something like "flashes their shiny security badge"

**Comment by GuyonBroadway** [comment]
You've added these but have not added any way to get them in game.

File: hippiestation/code/game/objects/items/badges.dm (hunk 3 above)
Comment (yoyobatty) [reply hippiestation/code/game/objects/items/badges.dm]: Agreed

**Comment by yoyobatty** [comment]
Not a big fan of the sprites and add a way of getting these ingame like guyon said.

**Comment by Lxtoofonea** [comment]
+1

**Comment by bardofrage** [comment]
+1

**Comment by ghost** [comment]
sprite looks shit tbh

**Comment by Carbonhell** [comment]
It has enough +1s, is the pr ready?

**Comment by DaydreamPuppeh** [comment]
It is not ready until he fixes the sprites.

**Comment by Carbonhell** [comment]
@AnthonyJNgo Once you replace the sprites and add a way to get them somehow, just ask a maintainer to reopen the PR.

**Pull request closed without merging**